cmd/collector/app/collector.go

```diff
@@ -101,26 +99,28 @@ func (c *Collector) Start(options *flags.CollectorOptions) error {
 
 	c.spanProcessor = handlerBuilder.BuildSpanProcessor(additionalProcessors...)
 	c.spanHandlers = handlerBuilder.BuildHandlers(c.spanProcessor)
-
+	// fmt.Printf("%v \n",options.GRPC.NetAddr.Endpoint)
 	grpcServer, err := server.StartGRPCServer(&server.GRPCServerParams{
-		HostPort:                options.GRPC.HostPort,
-		Handler:                 c.spanHandlers.GRPCHandler,
-		TLSConfig:               options.GRPC.TLS,
-		SamplingProvider:        c.samplingProvider,
-		Logger:                  c.logger,
-		MaxReceiveMessageLength: options.GRPC.MaxReceiveMessageLength,
-		MaxConnectionAge:        options.GRPC.MaxConnectionAge,
-		MaxConnectionAgeGrace:   options.GRPC.MaxConnectionAgeGrace,
+		Handler:          c.spanHandlers.GRPCHandler,
+		SamplingProvider: c.samplingProvider,
+		Logger:           c.logger,
+		ServerConfig: &configgrpc.ServerConfig{
+			NetAddr: confignet.AddrConfig{
+				Endpoint: options.GRPC.NetAddr.Endpoint,
+			},
+			TLSSetting:        options.GRPC.TLSSetting,
+			MaxRecvMsgSizeMiB: options.GRPC.MaxRecvMsgSizeMiB,
+			Keepalive:         options.GRPC.Keepalive,
+		},
```

Comment:
couldn't we just assign the whole struct?
```suggestion
		ServerConfig: options.GRPC,
```